go/private/go_repository.bzl

```diff
@@ -95,3 +112,5 @@ new_go_repository = repository_rule(
         ),
     },
 )
+
+new_go_repository = go_repository
```

Comment:
Might be good to have a comment explaining what `new_go_repository` was for future generations.